BD-2778 Add email capture form to back-in-stock notifs (#6826)
braze-inc/braze-docs: _docs/_partners/message_orchestration/channel_extensions/ecommerce/shopify/Shopify_Features/shopify_catalogs/back_in_stock.md

Use this Liquid tag  ``{% catalog_items <name_of_your_catalog> {{canvas_entry_properties.${catalog_update}.item_id}}`` at the top of your message, then use ``items[0].<field_name>` to access data about that item throughout the message.
{% endraw %}

## Connecting an email capture form

> If you have an email capture form to collect user interest in a specific product, you can connect the form to back-in-stock notifications and subscribe users to product updates. To do this, you’ll create a custom event after capturing a user’s email address. You'll then use the [`/users/track`]({{site.baseurl}}/api/endpoints/user_data/post_user_track#example-request-for-updating-a-user-profile-by-email-address) endpoint to update the user when you send the custom event. 

Reviewer: Is this referring to the IAM DND template? https://www.braze.com/docs/user_guide/message_building_by_channel/in-app_messages/drag_and_drop/templates/email_capture/ 

```suggestion
If you have an [email capture form]({{site.baseurl}}/user_guide/message_building_by_channel/in-app_messages/drag_and_drop/templates/email_capture/) to collect user interest in a specific product, you can connect the form to back-in-stock notifications and subscribe users to product updates. To do this, you’ll create a custom event after capturing a user’s email address. You'll then use the [`/users/track`]({{site.baseurl}}/api/endpoints/user_data/post_user_track#example-request-for-updating-a-user-profile-by-email-address) endpoint to update the user when you send the custom event. 
```

When you send Braze a `/users/track` request with only an email address, we will search for a user profile with that existing email address and subscribe the user. If we can’t find that email address, we’ll create a new user profile with only an email address so that they can receive messaging too.

### Step 1: Create a custom event to subscribe a user from updates

> You'll need these attributes in your custom event:

Reviewer: @rachel-feinberg does this mean they'll need custom event properties that map to these attributes?

Author: Correct. The attribute bullet points confirm it, but I can mention mapping above to make it clearer.

> You'll need these attributes in your custom event:

Author: ```suggestion
You'll need these custom event attributes to map to the catalog attributes:
```
@bre-fitzgerald Does this make more sense? 🤔

> You'll need these custom event attributes to map to the catalog attributes:

Author: ```suggestion
You'll need these custom event properties to map to the catalog attributes:
```

- `email_address`
- `product_id` with a value corresponding to the catalog item. If you have a different event property name like `product_name` or `product ID`, you can map those names to `product_id` in the Braze platform.
- `catalog_name` with a value corresponding to the catalog name. This is optional, but if you don’t have one you must set a default catalog for Braze to reference in the Braze platform.

### Step 2: Create a custom event to remove a user from updates (optional)

> If you don’t have a custom event to remove a user, users will automatically be removed after 90 days You'll need the same attributes in Step 1 to create this custom event.

Reviewer: Missing period
```suggestion
If you don’t have a custom event to remove a user, users will automatically be removed after 90 days. You'll need the same attributes in Step 1 to create this custom event.
```

> After you create your custom event, your email form capture tool will subscribe users to updates for that specific product.

Reviewer: If this is incorrect, feel free to disregard, but it seems odd to call this a tool here 🤔 
```suggestion
After you create your custom event, your email capture form will subscribe users to updates for that specific product.
```
